src/state/resolver.ts

```diff
@@ -47,51 +53,78 @@ export class EthStateResolver implements IStateResolver {
       return {
         latest: true,
         genesis: isGenesis,
-        state,
+        state: state,
         transitionTimestamp: 0,
       };
     }
 
-    if (contractState.toBigInt() !== state) {
-      const transitionInfo = await contract.getTransitionInfo(state);
-
-      if (transitionInfo[4].toBigInt() === 0n) {
-        throw new Error('transition info contains invalid id');
-      }
+    if (!contractState.id.eq(id)) {
+      throw new Error(`state was recorded for another identity`);
+    }
 
-      if (transitionInfo[0].toBigInt() === 0n) {
-        throw new Error('no information of transition for non-latest state');
+    if (!contractState.state.eq(state)) {
+      if (contractState.replacedAtTimestamp.eq(zeroInt)) {
+        throw new Error(`no information about state transition`);
       }
-
       return {
         latest: false,
-        state,
-        genesis: isGenesis,
-        transitionTimestamp: transitionInfo[0].toBigInt(),
+        genesis: false,
+        state: state,
+        transitionTimestamp: contractState.replacedAtTimestamp.toString(),
```

Comment:
why toString()

Reply:
Since `transitionTimestamp` can be a number or string.
Since `contractState.replacedAtTimestamp` is `eth.BigInt` type we can have an overflow for `number` type.

Reply:
timestamp is numer of second since 1970. we don't need string

Reply:
Yes, I now. But internal eth implementation uses bigint.

Reply:
Fixed